### PR TITLE
Feat : Added Authentication to video library

### PR DESCRIPTION
**Pull request by Omkar-Chaskar**
Added Authentication through API
functioning pages
-Log in
-Log out
-Sign Up Page
Home Page
-Single video page
(Hide Icons And buttons in user is not log In)

### CODE REVIEW

**Review by MonikaJhingan — APPROVED** [review]
The folder structure is good. 
Code is clean .
Well indented.
It should be merged.

**Review by dev-enforced — APPROVED** [review]
The code looks really well structured!! The live preview authentication functionality is also working correctly!! But didn't notice the toasts that were implemented in the code appearing on the screen which you have defined on successful and unsuccessful auth.

File: src/component/home.js [hunk 1]
```diff
@@ -25,28 +25,35 @@ export default function Home() {
...
+                            likeDispatch({
+                              type: "ADD_TO_LIKE",
+                              payload: video
+                            })
+                          }
+                        />
+                      )}
+                    </span>
+                  ) : (
+                    ""
```
Comment (ultronjs) [inline src/component/home.js]: Just a suggestion you could have add a empty fragment instead of empty string

**Review by ultronjs — APPROVED** [review]
Everything looks fine and well structured.
Just that suggestion added above
Good to merge

File: src/component/home.js (hunk 1 above)
Comment (Omkar-Chaskar) [reply src/component/home.js]: thanks i will do that

**Pull request merged**